Add customer docs for address restriction [TIC-related]

jmcarp opened the pull request:
TBD:
* Document (lack of) restriction on logs / doppler / ssh
* Document deployer account limitation

Shouldn't be merged until TIC SCR is accepted.

jmcarp committed:
Add customer docs for address restriction.

brittag commented on content/docs/apps/tic.md:
```diff
@@ -0,0 +1,14 @@
+
+To comply with the [TIC Initiative](https://www.dhs.gov/trusted-internet-connections), you can configure cloud.gov to restrict access to cloud.gov system services to a set of IP address ranges. This restriction prevents users from your email domain (e.g., `gsa.gov`) from accessing cloud.gov services unless their requests originate from one of your whitelisted IP address ranges.
+
+To configure IP address restriction, create a [support ticket](mailto:cloud-gov-support@gsa.gov?body=Email%20domain%3A%0A%0AEgress%20IP%20ranges%3A%0A%0AAgency%20CIO%3A%0A). Because address restriction applies to all cloud.gov users within your email domain, we will confirm your request with your agency CIO before changing configuration.
```
Trying to think of ways to make this easier - can we give them an email template that includes asking them to cc their agency CIO on the email request, which instructs the CIO to reply-all with approval?

jmcarp replied:
This is a (very minimal) email template that should render as:

```
Email domain:

Egress IP ranges:

Agency CIO:
```

We could add a note to the template text asking the user to cc their CIO. Help drafting that text would be appreciated 😃

brittag commented on content/docs/apps/tic.md:
```diff
@@ -0,0 +1,14 @@
+---
+menu:
+  docs:
+    parent: advanced
```
I think we need an Org Manager section of some kind ("Managing orgs"?), and to put this page in that section instead of in the docs sections about app development.

brittag commented on content/docs/apps/tic.md:
```diff
@@ -0,0 +1,14 @@
+menu:
+  docs:
+    parent: advanced
+title: Restricting access
```
Could this be more a slightly more specific title? "Restrict access by IP for TIC" ? I don't know if that's right, just something a bit more specific.

mogul replied:
I am thinking we want to split this into two docs... One will be about TIC compliance for cloud.gov in general, which then refers to this page for directions about configuring the "restrict client request origin for your domain" feature. That scopes this doc more tightly to what it is, while enabling us to give lots of hits for "TIC" in searches in the other doc.

brittag replied:
Maybe "Restrict administrative access by IP" ?

jmcarp committed:
Move TIC docs to new section.

mogul committed:
Smoothing it out, add route services note

mogul commented on content/docs/orgs/restricting-users-to-ip-ranges.md:
```diff
@@ -0,0 +1,19 @@
+We can configure cloud.gov to restrict users in a particular domain (e.g., `gsa.gov`) from using cloud.gov services unless their requests originate from a trusted network location. Your users will still be able to connect to cloud.gov from other locations, but any requests they make will be rejected. You can use this feature to help your agency comply with the [Trusted Internet Connections (TIC)  ...
+
+### Contact support to configure restrictions for your domain
+Create a [support ticket](mailto:cloud-gov-support@gsa.gov?body=Email%20domain%3A%0A%0AEgress%20IP%20ranges%3A%0A%0AAgency%20CIO%3A%0A) specifying the IP address ranges that are valid for your domain. Because address restriction applies to all cloud.gov users from your email domain, we will confirm your request with your agency CIO before changing the configuration.
```
This still needs work... We should put in a template that's clear enough that the CIO can be cc'd for signoff.

mogul committed:
Simpler language

mogul commented on content/docs/orgs/restricting-users-to-ip-ranges.md:
```diff
@@ -0,0 +1,19 @@
+We can configure cloud.gov to restrict users in a particular domain (e.g., `gsa.gov`) from using cloud.gov services unless their requests originate from a trusted network location. Your users will still be able to connect to cloud.gov from other locations, but any requests they make will be rejected. You can use this feature to help your agency comply with the [Trusted Internet Connections (TIC)  ...
+
+### Contact support to configure restrictions for your domain
+Create a [support ticket](mailto:cloud-gov-support@gsa.gov?body=Email%20domain%3A%0A%0AEgress%20IP%20ranges%3A%0A%0AAgency%20CIO%3A%0A) specifying the IP address ranges that are valid for your domain. Because address restriction applies to all cloud.gov users from your email domain, we will request confirmation from your agency CIO before changing the configuration.
```
Still needs work in the template. Customers should be able to cc an agency CIO and have them receive enough context to approve easily.

mogul committed:
Draft of the main TIC architecture docs

mogul committed:
smoothing out TIC interpretation

mogul committed:
Reordering for logical flow and more smoothing

mogul committed:
Merge branch 'master' into tic-docs

mogul committed:
Add mermaid diagrams, remove passive voice, add how to restrict app usage

Note that diagram titles are above the diagrams themselves and are not styled or centered in any way.

mogul committed:
Added intro for first case, and a better summary of route service

mogul committed:
Improved diagram title

brittag commented on content/docs/orgs/meeting-tic-requirements.md:
```diff
@@ -0,0 +1,100 @@
+
+### Restricting developer and operator access to cloud.gov services
+
+You can ensure that developer and operator access to cloud.gov services traverses your agency's TIC so that all manipulation of your application's deployment is monitored.
```
"manipulation of" -> "changes to" (manipulation sounds vaguely sinister)

mogul committed:
Streamlines text, includes IP filter in diagram

mogul committed:
Change position+style of diagram captions

brittag reviewed (dismissed):
Reviewed this with Bret

brittag commented on content/docs/orgs/restricting-users-to-trusted-ip-ranges.md:
```diff
@@ -0,0 +1,20 @@
+This feature only limits requests to cloud.gov services, such as the cloud.gov API accessed via the command line and the [cloud.gov dashboard](https://dashboard.fr.cloud.gov). This feature does not restrict requests directly to applications that you run on cloud.gov, though [you can do that](#restricting-access-to-your-own-applications) on your own.
+
+### Contact support to configure restrictions for your domain
+Create a [support ticket](mailto:cloud-gov-support@gsa.gov?body=Email%20domain%3A%0A%0AEgress%20IP%20ranges%3A%0A%0AAgency%20CIO%3A%0A) specifying the IP address ranges that are valid for your domain. Because address restriction applies to all cloud.gov users from your email domain, we will request confirmation from your agency CIO before changing the configuration.
```
@mogul Should we have a short note here that this isn't available _yet_ but will be available in the future?

Commit:
Add note that this is not yet available

brittag reviewed (approved):
Reviewed this and added a minor change (noting that it's not yet available in production)
